### ISSUE TITLE
[Bug] I think fp16 is not real when using TIMMBackbone

### ISSUE THREAD

**Issue by sleepfin** [open-mmlab/mmclassification#911]
### Describe the bug

When using TIMMBackbone, fp16 (or mixed precision) does not work

### To Reproduce

Use this config: https://github.com/open-mmlab/mmclassification/blob/master/configs/vision_transformer/vit-base-p16_ft-4xb544-ipu_in1k.py

1) change backbone to TIMMBackbone, use model_name='vit_small_patch16_224'
2) change head `in_channel` to 384

### Post related information

1) `auto_fp16_wrapper` (mmcv/runner/fp16_utils.py) is used when fp16(or mixed precision) training
2) `auto_fp16_wrapper` only convert input of model to fp16，but Parameter in model is still fp32, like `self.cls_token` in TIMMBackbone
3) In TIMMBackbone forward, `x = torch.cat((cls_token, x), dim=1)` before this code, x is fp16, after this code, x is fp32. All computation is fp32 after this code.
